onnx/defs/generator/defs.cc

```diff
@@ -523,4 +527,187 @@ ONNX_OPERATOR_SET_SCHEMA(
           updateOutputShape(ctx, 0, {batch_size, sample_size});
         }));
 
+static const char* Range_ver11_doc = R"DOC(
+Generate a tensor containing a sequence of numbers that begin at `start` and extends by increments of `delta` 
+up to `limit` (exclusive).)DOC";
+
+template <typename T>
+inline int compute_output_dim_for_range(
+    const TensorProto* start,
+    const TensorProto* limit,
+    const TensorProto* delta) {
+  const auto& start_data = ParseData<T>(start);
+  const auto& limit_data = ParseData<T>(limit);
+  const auto& delta_data =
+      delta != nullptr ? ParseData<T>(delta) : std::vector<T>(1, 1);
+
+  if (start_data.size() != 1 || limit_data.size() != 1 ||
+      delta_data.size() != 1) {
+    fail_shape_inference(
+        "Input to 'Range' op should be scalars (Tensor with only one element)");
+  }
+
+  int n = static_cast<int>(
```

Comment:
Please use int64, the largest allowed type in the spec below. Also, can we access the data during shape inference? I through this shape inference happens before we load the data.

Reply:
Yes, we can access data at shape inference if it is available as an initializer. There is a check on line 678 to make sure the necessary data is available as an initializer before invoking this method.

Reply:
And I don't see why it should be made int64. The value that is being returned is of `int` type (this is a dim value which is int type). So after applying the formule - ceil(....), it needs to be casted back to 'int' as per requirement. This looks valid to me.

Reply:
Can't `limit_data` be int64? I thought it can according to the allowed types of `limit_data`. If the dim type must be int32, please throw if overflow happens. You can do something like
```
int64_t n = ...
if (n > INT_MAX)
  throw ...
```

Reply:
ell I think 'int' may be either int64_t or int32_t based on the platform. So dim type can be either int32 or int64. Correct ? 

But I will add this check just to be safe when it is running on 32-bit platform...

Reply:
Dim value is int64 (see: https://github.com/onnx/onnx/blob/3ba2e31a4954b4c151dfcf7deb141f244a43fecf/onnx/onnx.in.proto#L451 ). I think we should use int64 here.

Reply:
I mean the return value, and the corresponding variable in the calling function should all be int64.

Reply:
Cool, so the only change we need is to use int64 in the entire line.